pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -369,155 +370,95 @@ func (s *GameServerSelector) matchLists(gs *agonesv1.GameServer) bool {
-	var causes []metav1.StatusCause
+func (s *GameServerSelector) Validate(fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
 
 	_, err := metav1.LabelSelectorAsSelector(&s.LabelSelector)
 	if err != nil {
-		causes = append(causes, metav1.StatusCause{
-			Type:    metav1.CauseTypeFieldValueInvalid,
-			Message: fmt.Sprintf("Error converting label selector: %s", err),
-			Field:   field,
-		})
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("labelSelector"), s.LabelSelector, fmt.Sprintf("Error converting label selector: %s", err)))
 	}
 
 	if runtime.FeatureEnabled(runtime.FeatureStateAllocationFilter) {
 		if s.GameServerState != nil && !(*s.GameServerState == agonesv1.GameServerStateAllocated || *s.GameServerState == agonesv1.GameServerStateReady) {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: "GameServerState value can only be Allocated or Ready",
-				Field:   field,
-			})
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("gameServerState"), *s.GameServerState, "GameServerState must be either Allocated or Ready"))
 		}
 	}
 
 	if runtime.FeatureEnabled(runtime.FeaturePlayerAllocationFilter) && s.Players != nil {
 		if s.Players.MinAvailable < 0 {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: "Players.MinAvailable must be greater than zero",
-				Field:   field,
-			})
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("players").Child("minAvailable"), s.Players.MinAvailable, apimachineryvalidation.IsNegativeErrorMsg))
 		}
 
 		if s.Players.MaxAvailable < 0 {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: "Players.MaxAvailable must be greater than zero",
-				Field:   field,
-			})
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("players").Child("maxAvailable"), s.Players.MaxAvailable, apimachineryvalidation.IsNegativeErrorMsg))
 		}
 
 		if s.Players.MinAvailable > s.Players.MaxAvailable {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: "Players.MinAvailable must be less than Players.MaxAvailable",
-				Field:   field,
-			})
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("players").Child("minAvailable"), s.Players.MinAvailable, "minAvailable cannot be greater than maxAvailable"))
 		}
 	}
 
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) && (s.Counters != nil || s.Lists != nil) {
-		causes = append(causes, metav1.StatusCause{
-			Type:    metav1.CauseTypeFieldValueInvalid,
-			Message: "Feature CountsAndLists must be enabled if Counters or Lists are specified",
-			Field:   field,
-		})
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("counters"), "Feature CountsAndLists must be enabled if Counters or Lists are specified"))
```

Comment:
How specific do you want the field path to be? Here there may be the case that "lists" is the field path.

Reply:
@igooch 
Yes, I only processed the counters field, the lists field should have been processed as well, I ignored it

thank you very much

Reply:
@igooch  done